lib/min_heap.rb

```diff
@@ -14,18 +14,24 @@ def initialize
   end
 
   # This method adds a HeapNode instance to the heap
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: o(logn)
+  # Space Complexity: o(logn)
   def add(key, value = key)
-    raise NotImplementedError, "Method not implemented yet..."
+    # raise NotImplementedError, "Method not implemented yet..."
+    @store << HeapNode.new(key, value)
+    heap_up(@store.length - 1)
   end
 
   # This method removes and returns an element from the heap
   #   maintaining the heap structure
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: o(logn)
+  # Space Complexity: o(1) i think
   def remove()
```

Comment:
Since `heap_down` is recursive this is O(log n) for time complexity just like `add`